packages/mockserver-config-writer/src/mockserver-config/package-json.ts

```diff
@@ -0,0 +1,157 @@
+    packageJson.devDependencies[mockserverModule] = version;
+    packageJson.ui5 = packageJson.ui5 || {};
+    packageJson.ui5.dependencies = packageJson.ui5.dependencies || [];
+    const ui5Dependencies = packageJson.ui5.dependencies.filter(
+        (dep) => dep !== '@sap/ux-ui5-fe-mockserver-middleware'
+    );
+    if (!ui5Dependencies.includes(mockserverModule)) {
+        ui5Dependencies.push(mockserverModule);
+    }
+    packageJson.ui5.dependencies = ui5Dependencies;
+}
+
+/**
+ * Get the tile name of the app from app id in manifest.json and flpSandbox.hml.
+ *
+ * @param appId - id of app from manifest.json
+ * @param flpSandboxContent - content of file flpSandbox.html as string
+ * @returns - name of the tile for the app, might be empty string if not found
+ */
+function getFlpSandboxTileName(appId: string, flpSandboxContent: string): string {
+    let flpSandboxTileName = '';
+    // Search for most recent tile name
+    const nameWithNs = `${appId.replace(/\./g, '')}-tile`;
+    if (flpSandboxContent.includes(nameWithNs)) {
+        flpSandboxTileName = nameWithNs;
+    } else {
+        // Search for older definitions of tile name
+        const nameNoNs = `${appId.split('.').pop()}-tile`;
+        if (flpSandboxContent) {
+            flpSandboxTileName = nameNoNs;
+        }
+    }
+    return flpSandboxTileName;
+}
+
+/**
+ * Returns whether the start-mock script in package.json is legacy.
+ *
+ * @param startMockScript - string of the start-mock script in package.json
+ * @returns - true: is a legacy script; false: not a legacy script, perhaps custom script
+ */
+function isLegacyStartMockScript(startMockScript: string): boolean {
+    return (
+        startMockScript.startsWith(`fiori run --config ./ui5-mock.yaml --open`) &&
+        startMockScript.includes('test/flpSandbox.html')
+    );
+}
+/**
+ * Add basic start script for mock server to package.json which does not open a specific file or tile.
+ * If there is an existing start-mock script, copy it to start-mock_.
+ *
+ * @param packageJson - content of package.json
+ */
+function enhanceBasicStartMockserverScript(packageJson: Package): void {
+    packageJson.scripts = packageJson.scripts || {};
+    if (
+        packageJson.scripts['start-mock'] &&
+        !packageJson.scripts['start-mock'].startsWith('fiori run --config ./ui5-mock.yaml')
+    ) {
+        packageJson.scripts['start-mock_'] = packageJson.scripts['start-mock'];
+    }
+    packageJson.scripts['start-mock'] = `fiori run --config ./ui5-mock.yaml --open \"/\"`;
+}
+
+/**
+ * Adds start-mock script to package.json, do nothing if script is up to date. If there is a
+ * custom start-mock script, copy it to start-mock_.
+ *
+ * @param fs - mem-fs reference to be used for file access
+ * @param packageJson - path to package.json
+ * @param webappPath - path to webapp folder
+ */
+function enhanceScripts(fs: Editor, packageJson: Package, webappPath: string): void {
```

Comment:
I would prefer a different way of adding the script. With your current you don't take into account what I as a user actually want as my script. I might have added additional things to my normal start script to e.g. start a server in the background or do xyz, therefore, I'd prefer is you copy the existing `start` script and find the existing `ui5 serve` or `fiori run` and then replace or add the `--config` parameter.

Reply:
Changed the logic:
- copy start -> start-mock script and add/replace `--config`
- fallback to basic script in case there is no start script, which should be an exceptional case